www/src/components/debounce-input.js

```diff
@@ -5,35 +5,52 @@ import PropTypes from "prop-types"
 class DebounceInput extends Component {
   static propTypes = {
     onChange: PropTypes.func.isRequired,
-    initialValue: PropTypes.string,
+    value: PropTypes.string,
     delay: PropTypes.number,
   }
 
   static defaultProps = {
-    initialValue: "",
+    value: ``,
     delay: 500,
   }
 
   state = {
-    value: this.props.initialValue,
+    inputValue: ``,
   }
 
-  onChangeText = e => {
-    this.setState({ value: e.target.value })
+  componentDidMount() {
+    this.setInputValue(this.props.value)
+  }
+
+  componentDidUpdate(prevProps, prevState, shouldFireRouteUpdate) {
```

Comment:
`prevState` and `shouldFireRouteUpdate` are unused - can we remove them?

Reply:
sure thing!